oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/JwtEncoder.java

```diff
@@ -37,16 +39,16 @@
 @FunctionalInterface
 public interface JwtEncoder {
 
-	// TODO: decide what the argument should be
+	// TODO: should the claims be a new type, or is a Map OK?
 
 	/**
-	 * Encodes the JWT from a {@link Jwt} to it's compact claims representation format.
+	 * Encodes the JWT from a set of claims to it's compact claims representation format.
 	 *
-	 * @param token the JWT value
-	 * @return a {@link Jwt} encoded to it's compact claims representation format
+	 * @param claims the JWT claims
+	 * @return a {@link Jwt}, its {@code tokenValue} containing its compact claims representation format
 	 * @throws JwtException if an error occurs while attempting to encode the JWT
 	 */
-	String encode(Jwt token) throws JwtException;
+	Jwt encode(Map<String, Object> claims) throws JwtException;
```

Comment:
It might give a more type-safe experience to do:

```java
String encode(Consumer<Jwt.Builder> jwtBuilderConsumer) throws JwtException
```

The other nice thing about this is that it makes it easier for the caller to override whatever defaults the `JwtEncoder` implementation sets since the implementation would apply the `Consumer` just before signing.

cc @jgrandja

Reply:
@jzheaux I don't think the `JwtEncoder` should set any default claims, those are going to come from the `ClientRegistration`, so a `Consumer` doesn't really feel intuitive to me here. For type safety we could create a `JwtClaimsSet` class, which could be used for the overrides as well in the `ClientRegistration`.